tests/cli/test_cli.py

```diff
@@ -0,0 +1,265 @@
+
+    alice_view_result = click_runner.invoke(nucypher_cli, alice_view_args, catch_exceptions=False, env=envvars)
+    assert alice_view_result.exit_code == 0
+    alice_view_response = json.loads(alice_view_result.output)
+
+    # Alice expresses her desire to participate in data sharing with nucypher
+    # by saving her public key somewhere Bob and Enrico can find it.
+    side_channel.save_alice_pubkey(alice_view_response['result']['alice_verifying_key'])
+
+    """
+    Scene 2: Bob installs nucypher, examines his configuration and expresses his
+    interest to participate in data retrieval by posting his public keys somewhere public (side-channel).
+    """
+    bob_init_args = ('bob', 'init',
+                     '--federated-only',
+                     '--network', TEMPORARY_DOMAIN,
+                     '--config-root', bob_config_root)
+
+    bob_init_response = click_runner.invoke(nucypher_cli, bob_init_args, catch_exceptions=False, env=envvars)
+    assert bob_init_response.exit_code == 0
+
+    # Alice uses her configuration file to run the character "view" command
+    bob_configuration_file_location = os.path.join(bob_config_root, BobConfiguration.CONFIG_FILENAME)
+    bob_view_args = ('--json-ipc',
+                     'bob', 'public-keys',
+                     '--config-file', bob_configuration_file_location)
+
+    bob_view_result = click_runner.invoke(nucypher_cli, bob_view_args, catch_exceptions=False, env=envvars)
+    assert bob_view_result.exit_code == 0
+    bob_view_response = json.loads(bob_view_result.output)
+
+    # Bob interacts with the sidechannel
+    bob_public_keys = MockSideChannel.BobPublicKeys(bob_view_response['result']['bob_encrypting_key'],
+                                                    bob_view_response['result']['bob_verifying_key'])
+
+    side_channel.save_bob_public_keys(bob_public_keys)
+
+    """
+    Scene 3: Alice creates a policy, and saves the policy metadata to a sidechannel.
+    """
+
+    random_label = random_policy_label.decode()  # Unicode string
+
+    derive_args = ('--mock-networking',
+                   '--json-ipc',
+                   'alice', 'derive-policy',
+                   '--config-file', alice_configuration_file_location,
+                   '--label', random_label)
+
+    derive_response = click_runner.invoke(nucypher_cli, derive_args, catch_exceptions=False, env=envvars)
+    assert derive_response.exit_code == 0
+
+    derive_response = json.loads(derive_response.output)
+    assert derive_response['result']['label'] == random_label
+
+    # Alice and the sidechannel: at Tinagre
+    policy = MockSideChannel.PolicyAndLabel(encrypting_key=derive_response['result']['policy_encrypting_key'],
+                                            label=derive_response['result']['label'])
+    side_channel.save_policy(policy=policy)
+
+    """
+    Scene 4: We catch up with Alice later on, after she has learned about existing Ursulas
+    """
```

Comment:
I'd add here that she grants access to Bob.

Reply:
:heavy_check_mark: